rfcs/0016-better-logging.md

```diff
@@ -0,0 +1,153 @@
+# Better Logging
+
+## Summary
+[summary]: #summary
+
+This RFC contains a proposal for a new logging system which will provide a better and more flexible format for our logging which can be easily processed by programs and is easy to construct into a human readable format.
+
+## Motivation
+[motivation]: #motivation
+
+Our logs in a state where they are both hard to read and hard to process. To break down the individual issues with our logs:
+
+- String escaping is broken
+- Messages contain lots of data in the form of large serialized sexp, reducing readability
+- No metadata is programmatically available without parsing the messages
+- Existing tools (jq, rq) are very slow at processing the raw format of our logs
+- Logs are very verbose, often containing redundant and unecessary information (long "paths", all logs are stamped with same info of host+pid which never changes across a single node)
```

Comment:
it might be useful to have this redundant info if you're looking at a line in isolation

not sure if that's a valid use case, though

Reply:
This information can be decorated externally though. Typically, tools like logstash would decorate that information for you. There are also on disk solutions we could discuss for organizing logs with some per log metadata.

Reply:
This information is useful in the context of integration tests where we run more than one "node" process on a single machine (host isn't useful here, but pid is)

Reply:
But if we add that to the metadata we should be fine